tests/src/core/testqgsadjacenttiles.cpp

```diff
@@ -0,0 +1,171 @@
+/***************************************************************************
+     testadjacenttiles.cpp
```

Comment:
What about putting them in the existing TestQgsMapRenderer test? 

(The only reason I'm reluctant to have a specific cpp test for this is that every new cpp test increases the build time considerably, and I'm concerned we'll eventually start timing out on Travis. So I'm in favour of cpp tests, but wary of having too many individual test files, if that makes sense!)